pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -607,19 +665,43 @@ func (r *ReconcileBareMetalHost) manageHostPower(prov provisioner.Provisioner, i
 		return actionContinue{provResult.RequeueAfter}
 	}
 
+	pendingRebootSince := info.host.Status.Provisioning.PendingRebootSince
+	poweredOnAt := info.host.Status.Provisioning.PoweredOnAt
+
+	desiredPowerOnState := info.host.Spec.Online
+	isInRebootProcess := poweredOnAt.Before(pendingRebootSince) || (!pendingRebootSince.IsZero() && poweredOnAt.IsZero())
+
+	if isInRebootProcess {
+		if !info.host.Status.PoweredOn {
+			if _, hasAnn := info.host.Annotations[rebootAnnotationPrefix]; hasAnn {
```

Comment:
This isn't checking for both forms of the annotation. Should it? See earlier comment about the utility function.

Reply:
It first checks for the suffixless. if exists - removes it, and later we check again if any reboot annotation exists. Since we removed the suffixless, we know that the only possible annotation is a suffixed one. This code is different from the original code of the PR. I think it might be confusing. see [this discussion](https://github.com/metal3-io/baremetal-operator/pull/424#discussion_r380390961) for more details.

Reply:
I'm not following. Where is only one of the annotations removed but the other is left behind? I think I see it removing either form on line 640, but I admit I'm having a little trouble following the logic across all of the different places this change touches the code.

Reply:
in line 677 the suffixless annotation being removed, if existed. When we reach line 686 we use `hasRebootAnnotation()` which returns true if either suffixless or suffixed annotation exist. Since we reached this line after we removed the suffixless, or more correctly - we reached this line after we checked that there's no suffixless annotation, so we know that if `hasRebootAnnotation()` returned `true` it means that a suffixed annotation exists.